tests/tests_thin_basic.yml

```diff
@@ -0,0 +1,138 @@
+---
+- name: Basic snapshot test
+  hosts: all
+  vars:
+    test_disk_min_size: "1g"
+    test_disk_count: 10
+    test_storage_pools:
+      - name: test_vg1
+        disks: "{{ range(0, 6) | map('extract', unused_disks) | list }}"
+        volumes:
+          - name: thin_pool_dev
+            thin_pool_name: thin_pool
+            thin_pool_size: '3g'
+            size: "1g"
+            thin: true
+          - name: lv2
```

Comment:
Is it enough to test with 1 thin pool?  There seems to be a problem using multiple thin pools with the storage subsystem in el7 - perhaps something to do with blivet on el7?  I get the following error:
```
Failed to commit changes to disk: Process reported exit code 3:   --virtualsize may not be zero.
  Run `lvcreate --help' for more information.
```
@vojtechtrefny @japokorn any ideas?  The storage role thin pool test uses only 1 volume - https://github.com/linux-system-roles/storage/blob/main/tests/tests_create_thinp_then_remove.yml#L37-L43 - which is why it passes on el7

If you really want to test with multiple volumes, then we'll need to define one list of volumes for el7, and another list for el8 and later.

Reply:
yes, I'll update the test to only use 1 pool.

I'll test multiple pools manually.

Reply:
@vojtechtrefny @japokorn it looks like if I add:

`thin_pool_size: '3g'`

whenever I allocate a volume that uses the thin pool it works as expected on RHEL 7 and Centos 7.

Reply:
> Is it enough to test with 1 thin pool? There seems to be a problem using multiple thin pools with the storage subsystem in el7 - perhaps something to do with blivet on el7?

We have quite an old version of blivet on EL7 so there could be a lot of bugs that were fixed between 7 and 8, but I don't remember anything specific about thin provisioning that could be causing this.

> The storage role thin pool test uses only 1 volume - https://github.com/linux-system-roles/storage/blob/main/tests/tests_create_thinp_then_remove.yml#L37-L43 - which is why it passes on el7

Looks like something we should add to our test suite.

>  it looks like if I add: thin_pool_size: '3g' whenever I allocate a volume that uses the thin pool it works as expected on RHEL 7 and Centos 7.

We have some logic in the storage role to calculate size of logical volumes and thin pools when not specified [here](https://github.com/linux-system-roles/storage/blob/main/library/blivet.py#L1772) so it's possible this is somehow broken with older blivet. Can you please check @japokorn?

I'll try to test this scenario with the storage role and I'll report an upstream issue for us to track this.

Reply:
Looks like the problem was actually invalid size for the second LV:

```
          - name: lv2
            thin_pool_name: thin_pool
            size: "15"
```

`size: "15"` is interpreted as 15 B and on CentOS 7 this is rounded down to 0 KiB so that's why lvcreate is complaining about `--virtualsize may not be zero`. With newer blivet on Fedora and CentOS 9/10 we automatically adjust this size to the minimal size which explains why it was failing only on CentOS 7.